cmd/nerdctl/build_test.go

```diff
@@ -402,3 +403,31 @@ CMD ["echo", "dockerfile"]
 	base.Cmd("build", "-t", imageName, buildCtx).AssertOK()
 	base.Cmd("run", "--rm", imageName).AssertOutExactly("dockerfile\n")
 }
+
+func TestBuildNoTag(t *testing.T) {
+	testutil.RequiresBuild(t)
+	testutil.DockerIncompatible(t)
+	base := testutil.NewBase(t)
+	dockerfile := fmt.Sprintf(`FROM %s
+CMD ["echo", "nerdctl-build-notag-string"]
+	`, testutil.CommonImage)
+	buildCtx, err := createBuildContext(dockerfile)
+	assert.NilError(t, err)
+	defer os.RemoveAll(buildCtx)
+
+	base.Cmd("--debug", "build", buildCtx).AssertOK()
+	imgs := base.Cmd("images", "--format", "json").OutLines()
+	imgs = imgs[:len(imgs)-1]
+	var noneImage *imagePrintable
+	for _, img := range imgs {
+		err := json.Unmarshal([]byte(img), &noneImage)
+		assert.NilError(t, err, img)
+		if noneImage.Tag == "<none>" {
+			break
+		}
```

Comment:
To check  that at least we have an image with tag  `<none>`  you have to provide a bool before the break
```
if noneImage.Tag == "<none>" {
                        nonTagExists = true
			break
		}
```

Reply:
Yes I got the point of `noneImage ` but it is more reader friendly with bool